src/main/java/org/apache/sling/graphql/core/engine/DefaultQueryExecutor.java

```diff
@@ -289,4 +332,71 @@ private String getDirectiveArgumentValue(Directive d, String name) {
             throw up;
         }
     }
+
+    GraphQLSchema getSchema(@NotNull String sdl, @NotNull Resource currentResource, @NotNull String[] selectors) {
+        readLock.lock();
+        String newHash = SHA256Hasher.getHash(sdl);
+        String resourceToHashMapKey = getCacheKey(currentResource, selectors);
+        String oldHash = resourceToHashMap.get(resourceToHashMapKey);
```

Comment:
might be good to add a comment about why you're using a composite cache key, that was initially unclear to me. IIUC that's because buildWiring uses the Resource and the services that it uses can changes outside of our control.